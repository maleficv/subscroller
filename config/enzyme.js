import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import fetch from 'isomorphic-fetch';

global.fetch = fetch;

Enzyme.configure({adapter: new Adapter()});