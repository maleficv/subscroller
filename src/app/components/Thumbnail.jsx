import React, {Component} from 'react';
import styled from 'react-emotion';

import ImageLinks from './ImageLinks';
import Media from './Media';

const Box = styled('div')`
  float: left;
  width: 33.33%;
  padding: 2px;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  
  @media only screen and (max-width: 960px) {
    width: 50%;
  }
  
  @media only screen and (max-width: 480px) {
    width: 100%;
  }
  
  &:hover {
    & > div {
      opacity: 1;
    }
  }
`;


class Thumbnail extends Component {
    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        return this.props.showLightbox({
            image: this.props.image,
            link: this.props.link
        })
    }

    render() {
        const {image, title, link} = this.props;
        const src = image.replace('gifv', 'mp4');
        return (
            <Box onClick={this.handleClick}>
                <ImageLinks title={title} image={src} link={link} hide={true} handleClick={this.handleClick}/>
                <Media src={src}/>
            </Box>
        )
    }
}

export default Thumbnail;