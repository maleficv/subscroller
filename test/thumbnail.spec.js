import React from 'react';
import {shallow} from 'enzyme';

import Thumbnail from '../src/app/components/Thumbnail';

describe('Single Thumbnail component', () => {
    const thumbnail = shallow(<Thumbnail image="https://i.imgur.com/sjHPrLE.jpg"/>);
    const img = thumbnail.find('img');

    test('Thumbnail matches snapshot', () => {
        expect(thumbnail).toMatchSnapshot();
    });

    test('Thumbnail has an image', () => {
        expect(img).toBeDefined();
    });
});