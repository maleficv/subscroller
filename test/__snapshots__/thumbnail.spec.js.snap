// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Thumbnail component Thumbnail matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Thumbnail
    image="https://i.imgur.com/sjHPrLE.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ImageLinks
          handleClick={[Function]}
          hide={true}
          image="https://i.imgur.com/sjHPrLE.jpg"
        />,
        <Media
          src="https://i.imgur.com/sjHPrLE.jpg"
        />,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleClick": [Function],
          "hide": true,
          "image": "https://i.imgur.com/sjHPrLE.jpg",
          "link": undefined,
          "title": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "src": "https://i.imgur.com/sjHPrLE.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ImageLinks
            handleClick={[Function]}
            hide={true}
            image="https://i.imgur.com/sjHPrLE.jpg"
          />,
          <Media
            src="https://i.imgur.com/sjHPrLE.jpg"
          />,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClick": [Function],
            "hide": true,
            "image": "https://i.imgur.com/sjHPrLE.jpg",
            "link": undefined,
            "title": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "src": "https://i.imgur.com/sjHPrLE.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
